app/src/test/java/ch/hikemate/app/authentication/AuthViewModelTest.kt

```diff
@@ -361,4 +362,50 @@ class AuthViewModelTest {
     // Verify that the onSuccess callback was called
     verify(mockOnSuccess).invoke()
   }
+
+  @Test
+  fun deleteAccount_calls_repository_deleteAccount_and_updates_currentUser_to_null() = runTest {
+    setupSignedInUser()
+
+    val mockOnSuccess: (() -> Unit) = mock()
+
+    // Simulate a successful account deletion by invoking the onSuccess callback
+    doAnswer { arguments ->
+          val onSuccess = arguments.getArgument<() -> Unit>(2)
+          onSuccess()
+          null
+        }
+        .`when`(mockRepository)
+        .deleteAccount(any(), any(), any(), any())
+
+    // Verify that currentUser is initially mockFirebaseUser
+    assertEquals(mockFirebaseUser, viewModel.currentUser.first())
```

Comment:
Question: Why do you use `.first()` here and not `value`? `.first()`, even if it might be working, doesn't look appropriated for a single user

Reply:
Actually, I took this from this file's others tests. I will change them all to `.value`. Nice catch!